### PR TITLE
docs: debugger guide

### PR DESCRIPTION
**Pull request by officialasishkumar**
# Pull Request Template

## Description
Added Debugger guide

Fixes [#1002](https://github.com/keploy/keploy/issues/1002)

## Type of change
Feature

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Additional Context (Please include any Screenshots/gifs if relevant)

...

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [ ] I have added corresponding tests.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by officialasishkumar** [comment]
@Sonichigo PTAL

**Comment by Sonichigo** [comment]
tagging @shivamsouravjha !! Can you add your review ?

File: versioned_docs/version-2.0.0/keploy-explained/debugger-guide.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            "mode": "auto",
+            "asRoot": true,
+            "console": "integratedTerminal",
+            "program": "main.go",
+            "args": [
+                "record",
+                "-c",
+                "<path_to_executable>"
+                // "--delay", "15"
+            ],
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/keploy-explained/debugger-guide.md:45]: delay can be removed from record

**Comment by shivamsouravjha** [comment]
Overall looks good to me just command for mac can be added as well

File: versioned_docs/version-2.0.0/keploy-explained/debugger-guide.md (hunk 1 above)
Comment (officialasishkumar) [reply versioned_docs/version-2.0.0/keploy-explained/debugger-guide.md]: done

**Comment by officialasishkumar** [comment]
> Overall looks good to me just command for mac can be added as well

Do you want me to add that to the example in args?

```- The `args` parameter represents the entire keploy command for running your application. For example, in [Gin + Mongo](https://keploy.io/docs/quickstart/samples-gin/), you run `keploy record -c "go run main.go handler.go"` where `record -c "go run main.go handler.go"` is the `"args"`.```

**Comment by shivamsouravjha** [comment]
1. What is mean is instead of just writing ctrl+shift+d you can write cmd +shift+d along (this is for macos)

**Comment by officialasishkumar** [comment]
@shivamsouravjha, updated!

**Comment by officialasishkumar** [comment]
Looks like the github actions are failing because of the checkout in the [workflow file](https://github.com/keploy/docs/actions/runs/7465674502/workflow?pr=292) which tries to checkout to the branch but couldn't find it. Is it the intended feature? @nehagup

**Comment by Sonichigo** [comment]
> Looks like the github actions are failing because of the checkout in the [workflow file](https://github.com/keploy/docs/actions/runs/7465674502/workflow?pr=292) which tries to checkout to the branch but couldn't find it. Is it the intended feature? @nehagup

Yes, because branch doesn't exit in the main keploy repo. This is done to maintain the branches, you can use main branch itself while raising pr (after merging it with your other feature branch)

File: versioned_docs/version-2.0.0/keploy-explained/debugger-guide.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+- The `args` parameter represents the entire keploy command to run your application. For example, in [Gin + Mongo](https://keploy.io/docs/quickstart/samples-gin/), on Linux, it's `record -c "go run main.go handler.go"`, and for MacOS, it's `record -c "docker compose up" --containerName "ginMongoApp"`.
+
+You can either add more objects in the "configurations" array or modify the "args" property within a single object to add more commands.
+
+> **Note**: By default, the keploy tests and reports will be generated in the keploy directory. You can change this by creating the [generate-config](https://keploy.io/docs/running-keploy/cli-commands/#generate-config) and specifying the desired path.
+
+
+## Running the Debugger
+
+Click the **Start Debugging** button or press **F5** to witness the magic of debugging unfold seamlessly.
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/keploy-explained/debugger-guide.md:77]: tried using f5 on macOS all it did was to turn of and on mic

File: versioned_docs/version-2.0.0/keploy-explained/debugger-guide.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  - debugger guide
+---
+
+# Debugger Guide 👨‍🔧
+
+In this blog, we will learn how to debug keploy using Visual Studio Code integrated debugger.  
+
+## Setting Up Configuration File
+
+To bring up the **Run and Debug** view, select the **Run and Debug** icon in the **Activity Bar** on the side of VS Code. You can also use the keyboard shortcut `Cmd+Shift+D`.
```
Comment (shivamsouravjha) [inline versioned_docs/version-2.0.0/keploy-explained/debugger-guide.md:17]: as I said something for both windows and macos (ctrl + shift +D and cmd+shift+D ) both

**Review by shivamsouravjha — APPROVED** [review]

**Comment by officialasishkumar** [comment]
Hey @Sonichigo, Can you please merge it if it's approved?

**Pull request merged**